pipenv/utils/resolver.py

```diff
@@ -550,24 +300,20 @@ def default_constraint_file(self):
 
     @property
     def pip_options(self):
```

Comment:
The past usage of :
```
if self._pip_options is None:

    ....
    
```
Suggest that this was called multiple times. Are we calling this multiple times still?
If so, could this be a cached property?

Reply:
It needs to be re-evaluated fresh each time.

Reply:
The problem with some of the past cached properties is they were leading to the need to have other bad code such as:  https://github.com/pypa/pipenv/pull/5793/files#diff-62e85156fc3b274046cc03dabfe6e3ceead503f330b98fb3611645216b4b6e1fL685

Reply:
![image](https://github.com/pypa/pipenv/assets/479892/4d13b0dc-8dbe-4be4-8949-7e3a562fb031)